src/System.Net.Http/src/System/Net/Http/Managed/DecompressionHandler.cs

```diff
@@ -47,31 +47,14 @@ public DecompressionHandler(DecompressionMethods decompressionMethods, HttpMessa
 
             HttpResponseMessage response = await _innerHandler.SendAsync(request, cancellationToken).ConfigureAwait(false);
 
-            while (true)
+            ICollection<string> contentEncoding = response.Content.Headers.ContentEncoding;
+            if (GZipEnabled && contentEncoding.Contains(s_gzip))
```

Comment:
Contains is misleading here. The content-encoding header is ordered, you can only decompress if it's the outermost encoding. From the spec I think the outermost is the last value.
"the sender that applied the encodings MUST generate a Content-Encoding
   header field that lists the content codings in the order in which
   they were applied"
https://tools.ietf.org/html/rfc7231#section-3.1.2.2

That said, there's rarely more than one content-encoding value so you could just skip decompression if there is.

Also, https://github.com/dotnet/corefx/blob/25b0a9b6f784216de7f357ba680aa9a835ad3cba/src/System.Net.Http/src/System/Net/Http/Managed/DecompressionHandler.cs#L102 seems to think it's the first value, not the last.

Reply:
If I'm reading the code correctly, Contains is effectively what the .NET Framework does.  But I've changed it to use the last encoding.